src/libs/itinerary/__tests__/useDistance.test.ts

```diff
@@ -0,0 +1,40 @@
+import { mocked } from 'ts-jest/utils'
+
+import { useGeolocation } from 'libs/geolocation'
+import {
+  emptyGeolocationContext,
+  geolocalisationContext,
+} from 'libs/geolocation/fixtures/geolocationContext'
+import { useDistance } from 'libs/geolocation/hooks/useDistance'
+import { formatDistance } from 'libs/parsers'
+
+jest.mock('libs/parsers')
+jest.mock('libs/geolocation')
+const mockedUseGeolocation = mocked(useGeolocation, true)
+
+describe('useDistance', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  const offerPosition = { lat: 31, long: 56 }
+
+  it('should call useGeolocation and formatDistance when geolocation is on', () => {
+    useDistance(offerPosition)
+    expect(useGeolocation).toHaveBeenCalledWith()
+    expect(formatDistance).toHaveBeenCalledWith(offerPosition, geolocalisationContext.position)
+  })
+
+  it('should return undefined when no offerpositon given', () => {
+    // @ts-expect-error offer position should not be undefined or null
+    useDistance()
```

Comment:
```suggestion
    useDistance({lat: undefined, lng: undefined})
```

comme ça pas besoin de // @ts-expect-error

Reply:
Oui mais ça n'est pas ce que je peux tester, dans la méthode, on teste si l'input est undefined, pas les clés...

Reply:
ah dans ce cas la méthode est fausse.
typescript check déjà qu'on appele `useDistance` avec un object du type 
```typescript
{
  lat?: number | null
  lng?: number | null
}
```

donc là on check `!offerPosition` mais on devrait plutôt checker si `typeof offerPosition.latitude !== 'number' || typeof offerPosition.longitude !== 'number'`

T'en penses quoi ?

Reply:
Ce cas est déjà géré dans getHumanizeRelativeDistance
Je pense qu'ici le check est pour se protéger du cas où on reçoit une données undefined (malgré le typage), ça ne devrait pas arriver mais bon...

Reply:
J'imagine pour les geoloc qui viennent des hist qui viennent de useSearchInfiniteQuery ? 
A mon avis il faudrai mettre plus de la validation de données à la réception, mais c'est un autre combat

Reply:
C'est vrai que tout laisse à croire que c'est pas fait exprès https://github.com/pass-culture/pass-culture-app-native/commit/702566442fb1f7f9eb3aa89cff2bf2bb0b286601 ...
Je te propose qu'on laisse comme c'est, au cas où ça serve de se protéger des données venant de l'API qu'on n'ont pas été validées, et qu'on ajoute un TODO